scripts/inject-deterministic.py

```diff
@@ -1,50 +1,171 @@
+            headers["Content-Security-Policy"] = "; ".join(directives)
+            ctx.log.info("Updated  Header %s \n" % headers["Content-Security-Policy"])
+
+        return headers
+
     def response(self, flow):
+
         millis = int(round(time.time() * 1000))
 
         if "content-type" in flow.response.headers:
-
             if 'text/html' in flow.response.headers["content-type"]:
                 ctx.log.info("Working on {}".format(flow.response.headers["content-type"]))
 
                 if "charset=" in flow.response.headers["content-type"]:
                     encoding = flow.response.headers["content-type"].split("charset=")[1]
                 else:
-                    encoding = "utf-8"
+                    encoding = "latin-1"
 
                 flow.response.decode()
                 html = bytes(flow.response.content).decode(encoding)
 
                 with open("scripts/catapult/deterministic.js", "r") as jsfile:
                     js = jsfile.read().replace("REPLACE_LOAD_TIMESTAMP", str(millis))
+
+                    # remove comments from js file
+                    js = re.sub(re.compile("//.*?\n"), "", js)
+
                     if js not in html:
-                        scriptIndex = re.search('(?i).*?<head.*?>', html)
-                        if scriptIndex is None:
-                            scriptIndex = re.search('(?i).*?<html.*?>', html)
-                        if scriptIndex is None:
-                            scriptIndex = re.search('(?i).*?<!doctype html>', html)
-                        if scriptIndex is None:
+                        script_index = re.search('(?i).*?<head.*?>', html)
+                        if script_index is None:
+                            script_index = re.search('(?i).*?<html.*?>', html)
+                        if script_index is None:
+                            script_index = re.search('(?i).*?<!doctype html>', html)
+                        if script_index is None:
                             ctx.log.info("No start tags found in request {}. Skip injecting".format(flow.request.url))
                             return
+                        script_index = script_index.end()
+
+                        nonce = None
+
+                        if flow.response.headers.get("Content-Security-Policy", False):
+                            nonce = self.get_nonce_from_headers(flow.response.headers)
+                            ctx.log.info("nonce : %s" % nonce)
 
-                        scriptIndex = scriptIndex.end()
+                            if self.get_csp_script_sources(flow.response.headers) and not nonce:
+                                # generate sha256 for the script
+                                hash_object = hashlib.sha256(js.encode('utf-8'))
+                                script_sha256 = "{}".format(
+                                    base64.b64encode(hash_object.digest())
```

Comment:
Use `hexdigest` here so you have a string instead of a bytes object, we shouldn't be using bytes and then stripping b''

Reply:
b64encode requires a byte object 
https://docs.python.org/2/library/base64.html#base64.b64encode

Reply:
Used base64.b64encode(hash_object.digest()).decode("utf-8")